### PR TITLE
sql: use embedded certs

### PR DESCRIPTION
**Pull request by tamird**
Fixes nightly sqllogictest.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3310)

### CODE REVIEW

**Comment by mberhault** [comment]
have you verified that a standalone sqllogictest binary works?

File: sql/logic_test.go [hunk 1]
```diff
@@ -204,27 +204,25 @@ func (t *logicTest) setUser(tempDir, user string) {
...
+	caPath := filepath.Join(security.EmbeddedCertsDir, "ca.crt")
+	certPath := security.ClientCertPath(security.EmbeddedCertsDir, user)
+	keyPath := security.ClientKeyPath(security.EmbeddedCertsDir, user)
 
-	// `github.com/lib/pq` requires that private key file permissions are
-	// "u=rw (0600) or less".
-	tmpKeyPath := tempRestrictedCopy(t.T, keyPath, tempDir)
+	// Copy these assets to disk from embedded strings, so this test can
+	// run from a standalone binary.
+	tempCAPath, _ := tempRestrictedCopy(t.T, tempDir, caPath, "TestLogic_ca")
```
Comment (petermattis) [inline sql/logic_test.go]: Rather than passing in the prefix for the temp file, you could derive it from the source path using `filepath.Base()`.

**Comment by petermattis** [comment]
LGTM, though we might want something that requires a bit less boilerplate per test since other tests might need this in the future when we move to using pgwire more widely. I'm thinking of the acceptance tests as one example.

File: sql/logic_test.go (hunk 1 above)
Comment (tamird) [inline sql/logic_test.go]: That's true, but in this case it's tagged with the test name, so I'd rather leave it as-is.

Comment (petermattis) [inline sql/logic_test.go]: I don't see why that matters, but ok, this isn't very important.

**Comment by tamird** [comment]
I think this is pretty close to the minimum boilerplate that can be had. We can export `tempRestrictedCopy` for other tests, but that's about it.

**Comment by petermattis** [comment]
I disagree about this being the minimum boilerplate. There could be a single function which creates a temporary directory and writes the embedded certs out to it with the appropriate permissions and returns the lib/pq parameters already encoded.

**Comment by tamird** [comment]
Oh, sure, if you wanted to reduce the boilerplate all the way out to getting a lib/pq-specific thing.

**Comment by tamird** [comment]
@mberhault testing this locally resulted in serious static-binary-related sadness. I'm still trying to get to the bottom of it with a minimal repro, but for now `lib/pq` calls `user.Current()` which panics in a static sql.test binary.

**Comment by tamird** [comment]
https://github.com/golang/go/issues/13470

**Comment by tamird** [comment]
OK, So I think the segfault issue now only affects PGWire test, which we don't run in a static binary, and logic test does pass in the static binary in my local docker container. I'm going to merge this and hope for the best.

**Comment by tschottdorf** [comment]
LGTM

**Comment by petermattis** [comment]
@tamird FYI, I just noticed that the benchmarks in `sql/bench_test.go` are broken due to using non-SSL connections.

**Comment by tamird** [comment]
I'm about to get on a plane, but yeah that seems bad. Might be tricky to fix since local postgres will need certs. I'll take a look in the air.

**Pull request merged**